app/controllers/Challenge.scala

```diff
@@ -136,20 +136,27 @@ final class Challenge(
   def decline(id: String) =
     Auth { implicit ctx => _ =>
       OptionFuResult(api byId id) { c =>
-        if (isForMe(c)) api decline c
+        if (isForMe(c)) api decline(c, None)
         else notFound
       }
     }
   def apiDecline(id: String) =
-    Scoped(_.Challenge.Write, _.Bot.Play, _.Board.Play) { _ => me =>
-      api.activeByIdFor(id, me) flatMap {
-        case None =>
-          env.bot.player.rematchDecline(id, me) flatMap {
-            case true => jsonOkResult.fuccess
-            case _    => notFoundJson()
+    ScopedBody(_.Challenge.Write, _.Bot.Play, _.Board.Play) { implicit req => me =>
+      env.setup.forms.api.declineChallenge
+        .bindFromRequest()
+        .fold(
+          err => BadRequest(apiFormError(err)).fuccess,
```

Comment:
Not sure if we should return `400` or just ignore the error and proceed here. `400` has the chance of breaking consumers that (wrongly) send completely empty JSON requests (no squiggly brackets in the body).